staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -264,17 +264,26 @@ func (p *jsonPatcher) computeStrategicMergePatch(currentObject runtime.Object, _
 }
 
 func (p *jsonPatcher) firstPatchAttempt(currentObject runtime.Object, currentResourceVersion string) (runtime.Object, error) {
-	// first time through,
-	// 1. apply the patch
-	// 2. save the original and patched to detect whether there were conflicting changes on retries
+	var err error
 
-	objToUpdate := p.restPatcher.New()
+	// Encode will convert & return a versioned object in JSON.
+	// Store this JS for future use.
+	p.originalObjJS, err = runtime.Encode(p.codec, currentObject)
+	if err != nil {
+		return nil, err
+	}
 
-	originalJS, patchedJS, err := p.patchObjectJSON(currentObject, objToUpdate)
+	// Apply the patch. Store patched result for future use.
+	p.originalPatchedObjJS, err = p.patchObjectJSON(p.originalObjJS)
```

Comment:
prefer not mutating until we're sure we didn't hit an error

Reply:
Same here--no conflict errror, no retry.